packages/xo-server/src/api/vm.js

```diff
@@ -1159,12 +1187,77 @@ resume.resolve = {
 
 // -------------------------------------------------------------------
 
-export async function revert({ snapshot }) {
+export const revert = defer(async function($defer, { snapshot }) {
   await this.checkPermissions(this.user.id, [
```

Comment:
A self user will always be an admin of the VM so it shouldn't be a problem.

Reply:
Yes, i asked this question because i saw that you handled this case [here](https://github.com/vatesfr/xen-orchestra/pull/3693/files#diff-822e0f553f3965578daa87822173851eR805).